File: E-Shop/Back-End/Project.Data/Dto/Product/GetProductDto.cs
using Project.Data.Enities;

namespace Project.Data.Dto.Product;
public class GetProductDto {
    public Guid Id { get; set; }
    public string Name { get; set; }
    public double Price { get; set; }
    public double? Discount { get; set; }
    public int Stock { get; set; }
    public string Picture { get; set; }
    public ProductTypes ProductType { get; set; }
}

File: E-Shop/Front-End/ProjectFront/src/app/models/Product/GetProductDto.ts
import { Guid } from "guid-typescript";
import { ProductTypes } from "src/app/enums/ProductType";

export class GetProductDto{
    id!: Guid;
    name!: string;
    price!: number;
    discount!: number;
    stock!: number;
    picture!: string;
    productType?: ProductTypes
}